File: caesar.py
# Caesar Cipher

# Setup
alphabet = 'abcdefghijklmnopqrstuvwxyz'

# Encryption
def encrypt(plaintext, key):
    ciphertext = ''
    for letter in plaintext:
        if letter.lower() in alphabet:
            is_upper = False
            if letter not in alphabet:
                is_upper = True
            letter = letter.lower()
            encrypted_letter = alphabet[(alphabet.index(letter) + key) % 26]
            if is_upper == True:
                encrypted_letter = encrypted_letter.upper()
            ciphertext += encrypted_letter
        else:
            ciphertext += letter
    return ciphertext

# Decryption
def decrypt(ciphertext, key):
    plaintext = ''
    for letter in ciphertext:
        if letter.lower() in alphabet:
            is_upper = False
            if letter not in alphabet:
                is_upper = True
            letter = letter.lower()
            decrypted_letter = alphabet[(alphabet.index(letter) - key) % 26]
            if is_upper == True:
                decrypted_letter = decrypted_letter.upper()
            plaintext += decrypted_letter
        else:
            plaintext += letter
    return plaintext

# Crack
def crack(ciphertext):
    for version in range(0, 26):
        cracked = ''
        for letter in ciphertext:
            if letter.lower() in alphabet:
                is_upper = False
                if letter not in alphabet:
                    is_upper = True
                letter = letter.lower()
                decrypted_letter = alphabet[(alphabet.index(letter) + version) % 26]
                if is_upper == True:
                    decrypted_letter = decrypted_letter.upper()
                cracked += decrypted_letter
            else:
                cracked += letter
        print("+" + str(version) + " " + cracked)

File: caesar.js

const alphabet = "abcdefghijklmnopqrstuvwxyz";

function encrypt(plaintext, key) {
    // Defining the variables
    let ciphertext = "";

    // Encryption
    for (iterator = 0; iterator < plaintext.length; iterator++) {
        let currentCharacter = plaintext.charAt(iterator);
        let isUpper = false;
        if (alphabet.includes(currentCharacter.toLowerCase()) == true) {

            // Checking if the current letter is uppercase
            if (alphabet.includes(currentCharacter) == false) {
				isUpper = true;
			}

            currentCharacter = currentCharacter.toLowerCase();
            oldIndex = alphabet.indexOf(currentCharacter);
            newIndex = (oldIndex + key) % 26;
            encryptedLetter = alphabet.charAt(newIndex);


            if (isUpper == true) {
                encryptedLetter = encryptedLetter.toUpperCase();
            }

            ciphertext += encryptedLetter;
        }

        else {
			ciphertext += currentCharacter;
		}
    }
    console.log(ciphertext);
}

function decrypt(ciphertext, key) {
    // Defining the variables
    let plaintext = "";

    // Decryption
    for (iterator = 0; iterator < ciphertext.length; iterator++) {
        let currentCharacter = ciphertext.charAt(iterator);
        let isUpper = false;
        if (alphabet.includes(currentCharacter.toLowerCase()) == true) {

            // Checking if the current letter is uppercase
            if (alphabet.includes(currentCharacter) == false) {
				isUpper = true;
			}

            currentCharacter = currentCharacter.toLowerCase();
            oldIndex = alphabet.indexOf(currentCharacter);
            newIndex = oldIndex - key;
            while (newIndex < 0) {
				newIndex += 26;
			}
            decryptedLetter = alphabet.charAt(newIndex);


            if (isUpper == true) {
                decryptedLetter = decryptedLetter.toUpperCase();
            }

            plaintext += decryptedLetter;
        }

        else {
			plaintext += currentCharacter;
		}
    }
    console.log(plaintext);
}


